src/pip/_internal/req/req_install.py

```diff
@@ -252,6 +254,13 @@ def match_markers(self, extras_requested: Optional[Iterable[str]] = None) -> boo
         else:
             return True
 
+    @property
+    def has_ignore_dependencies(self) -> bool:
+        if isinstance(self.comes_from, InstallRequirement):
+            return self.comes_from.ignore_dependencies
+        else:
+            return self.ignore_dependencies
```

Comment:
This feels a bit too prone to error; a caller would get incorrect behaviour if `req.ignore_dependencies` is used instead of `req.has_ignore_dependencies`. It may be better to override `ignore_dependencies directly in `__init__` instead?

```python
if ignore_dependencies:
    self.ignore_dependencies = ignore_dependencies
elif isinstance(comes_from, InstallRequirement):
    self.ignore_dependencies = comes_from.ignore_dependencies
else:
    self.ignore_dependencies = False
```

(I believe the more common idoim is to make the explicitly passed parameter override those inferred from other values.)

Reply:
I did it because of mypy failing, but now there are no errors..